sdcm/utils/common.py

```diff
@@ -714,6 +715,33 @@ def clean_instances_aws(tags_dict, dry_run=False):
                 LOGGER.debug("Done. Result: %s\n", response['TerminatingInstances'])
 
 
+def remove_test_runner_on_success(runner_ip: Optional[str],
+                                  backend: Optional[str],
+                                  region: Optional[str]):
+    def _remove_aws_test_runner(region: str, runner_ip: str):
+        client = boto3.client('ec2', region_name=region)
+        instance_filter = [{"Name": "ip-address", "Values": [runner_ip]}]
+        desc_result = client.describe_instances(Filters=instance_filter)
+        runner_id = desc_result["Reservations"][0]["Instances"][0]["InstanceId"]
+        result = client.terminate_instances(InstanceIds=[runner_id])
+        LOGGER.info("Sct runner removal result: %s", result)
+
+    def _remove_gce_test_runner(region: str, runner_ip: str):
+        driver = get_gce_service(region)
+        instances = list_instances_gce(tags_dict={"NodeType": "sct-runner"})
+        runner_instance = [item for item in instances if runner_ip in item.public_ips][0]
+        status = driver.destroy_node(runner_instance)
+        LOGGER.info("Sct runner removal status: %s", status)
+
+    if backend == 'aws':
+        _remove_aws_test_runner(region=region, runner_ip=runner_ip)
+
+    if backend == 'gce':
+        _remove_gce_test_runner(region=region, runner_ip=runner_ip)
+
+    LOGGER.info("Done.")
```

Comment:
This function doesn't check keep-alive tags. So, again: think about reusing existing code.

Reply:
Yea, the idea here is that we don't care about keep-alives when the tests succeed. If it didn't this step will be skipped either way.

Reply:
It's not true. We have two types of keep-alive: unconditional and keep-on-failure. So, we still need to check it for the unconditional case.

Reply:
Yep, moved away from a separate new function to reusing and slightly extending `clean_sct_runners()`. If there is "alive" in the `keep` tag, we won't terminate the instance, if there is a conditional `keep: <number>` tag and the test was successful, we'll terminate the runner